apps/docs/src/guide/predicates/index.md

```diff
@@ -6,22 +6,32 @@
 
 # Predicates
 
-Predicates, in Sway, are programs that return a Boolean value, and they do not have any side effects (they are pure).
+Predicates in Sway are specific types of programs that return a boolean value. Since they are pure functions, they do not produce side effects.
```

Comment:
```suggestion
Predicates are a Sway program type that return a boolean value. Since they are pure functions, they do not produce side effects.
```

Reply:
Should we be making some more reference to what predicates _are_? Or do we expect people to be aware of their characteristics and have come from a Sway realm? Sometimes there seem to be some confusion on this when people have come straight to the TS SDK or not delved into program types.

Something along the lines of: Predicates represent conditions that need to be satisfied for a transaction to be considered valid. A predicate is executed off-chain until validated, in which the transaction will be executed.  This reduces congestion and gas costs for carrying out similar computations on-chain.

Reply:
I agree we could expand on it, but the suggested paragraph sounds complex/abstract to me. 🤔 

We could draw inspiration from the other docs and also add links to both at the end of the page:
 - https://fuellabs.github.io/sway/v0.39.0/book/sway-program-types/predicates.html
 - http://rust.fuel.network/v0.41.0/getting-started/predicates.html

Reply:
@danielbate @arboleya could you please provide your opinion on the description in this commit? I would appreciate your feedback:

https://github.com/FuelLabs/fuels-ts/pull/1012/commits/3ca35392970a24cc156a572e0fd53b3a49a26b00#diff-58e11a4f3fa5238fff5b1b250ae6d34637d64e9f1a403ffbd27ec046421c959eR9-R13

Reply:
Much better! ⚡ 

Final considerations:

 - Perhaps we could have a small "Debugging" section summarizing [this](https://fuellabs.github.io/sway/v0.39.0/book/sway-program-types/predicates.html#debugging-predicates) (or a link to it)
 - IIRC, Predicates can't read state and only operate based on the parameters they receive. Is that correct? If yes, this is another thing we could mention.
 - After all of this, see if there's any value in adding the links to the other docs (Sway, Rust SDK) at the end of the page or where relevant.